gapir/cc/linux/gles_renderer.cpp

```diff
@@ -121,9 +122,17 @@ GlesRendererImpl::GlesRendererImpl(GlesRendererImpl* shared_context)
         , mSharedContext(shared_context != nullptr ? shared_context->mContext : 0)
         , mPbuffer(0) {
 
-    mDisplay = XOpenDisplay(nullptr);
-    if (mDisplay == nullptr) {
-        GAPID_FATAL("Unable to to open X display");
+    if (shared_context != nullptr) {
+        // Ensure that shared contexts also share X-display.
+        // Drivers are know to misbehave/crash without this.
+        mDisplay = shared_context->mDisplay;
+        mOwnsDisplay = false;
```

Comment:
So are shared contexts have an exclusive single owner? I thought they were shared - and would require refcounts?